### PR TITLE
Toplevel BP <--> HB test

### PR DESCRIPTION
**Pull request by sripathi-muralitharan**
This PR adds a SPMD program and the corresponding BP code that performs matrix-matrix add on a 2x2 tile group in a single pod of the manycore.

### CODE REVIEW

**Comment by tommydcjung** [comment]
Can you make your changes in nbf.py more maintainable by separating out bp-specific codes into a separate class/file (e.g. nbf_bp.py), which uses NBP as an abstract class?

**Review by tommydcjung — CHANGES_REQUESTED** [review]
requesting nbf.py refactor

**Comment by dpetrisko** [comment]
@sripathi-muralitharan

File: software/mk/Makefile.builddefs [hunk 1]
```diff
@@ -295,8 +296,24 @@ $(BSG_MANYCORE_LIB): $(BSG_MANYCORE_LIB_OBJS)
...
+NBF_SCRIPT   ?= $(NBF_PY)
+BP_MEM_IMAGE ?=
+BP_DRAM_BASE ?=
+BP_DRAM_POD  ?=
+endif
+
 %.nbf: %.riscv
-	python $(NBF_PY) $*.riscv \
+	python $(NBF_SCRIPT) \
+	  $*.riscv \
```
Comment (tommydcjung) [inline software/mk/Makefile.builddefs:316]: Can you make these parameters a separate makefile variable (e.g. $NBF_PARAM), and add the bp specific params if $HOST equals bp? so it doesn't break the existing nbf.py

File: software/spmd/bp_manycore_test/bp_main.c [hunk 2]
```diff
@@ -0,0 +1,164 @@
+#include <stdint.h>
```
Comment (tommydcjung) [inline software/spmd/bp_manycore_test/bp_main.c:1]: Add bp_manycore_test to NO-RECURSE. https://github.com/bespoke-silicon-group/bsg_manycore/blob/eb132d1915350ac79df451b1a7bd37caf5874ce3/software/spmd/Makefile#L9

File: software/mk/Makefile.builddefs (hunk 1 above)
Comment (sripathi-muralitharan) [reply software/mk/Makefile.builddefs:316]: I can do that for the BP-specific parameters. But which script to run is going to remain a separate variable as it is now.

**Review by tommydcjung — APPROVED** [review]

**Pull request merged**